### DISCUSSION TITLE
Tray icon to replace the 'updates' nwg-panel executor

### DISCUSSION THREAD

**Discussion by nwg-piotr** [nwg-piotr/nwg-shell#55] [Porting for other distros]
All the info below concerns the not-yet-merged [indicator](https://github.com/nwg-piotr/nwg-shell-config/tree/indicator) nwg-shell-config branch.

As soon as it's merged, it'll provide a system tray indicator, that will take care of checking and installing updates. It means, that the Arch-specific 'updates' nwg-shell executor goes on retirement, and will no longer be shipped in preinstalled panel configs. @mac-a-r0ni (and possibly other packagers): you will no longer need to remove it from default panel configs.

The `updater_indicator.py` script is meant to work with various Linux distributions. It provides the `nwg-update-indicator` command. For now just Arch-related stuff is implemented.

What does it do?

1. Checks the Linux distribution;
2. Checks updates availability (on startup and then periodically);
3. Runs the `nwg-system-update` script, to install updates.

The form below allows users to use their own command instead of `nwg-system-update`. Those may be e.g. pamac-addicted Arch derivatives users. ;)

The provided `nwg-system-update` script is very simple in the Arch version, and uses `baph` to install updates:

```bash
#!/usr/bin/env bash
# Packagers: you need to replace this script with something, that will work for your distro.

baph -u &&
echo $1; read;
swaymsg reload
```

(`$1` carries localized "Press enter to exit...")

The [updater_indicator.py](https://github.com/nwg-piotr/nwg-shell-config/blob/indicator/nwg_shell_config/update_indicator.py) code is quite well documented inside. To add support for your Linux distribution, please take a look at them. You may either contribute to the code, or just tell me what to do.

![image](https://user-images.githubusercontent.com/20579136/213604719-91ad0623-8a77-4c49-a00c-875f46af320e.png)

![image](https://user-images.githubusercontent.com/20579136/213678730-2d5aa756-40cd-41a5-ade2-cf86c1c5e8a3.png)

**Comment by mac-a-r0ni** [comment]
This is excellent news! Thank you for extending this functionality! I will work out updates and try and create a update script for Slackware within the next day or so, can't wait to try it out!

**Reply by nwg-piotr** [reply]
I'm glad you like the idea. This is especially for you, and possibly other distro packagers.

By the way: this is undocumented, but sending `USR1` to `nwg-update-indicator` triggers `check-updates`. Maybe you'll find it useful in your scripts.
